File: lang/go/test/muse/algorithms/search/BinarySearch/BinarySearchTest.go
package main

import (
    "muse/algorithms/search/BinarySearch"
    "muse/util/SequenceBuilder"
    "muse/util/TestRunner"
)

func testBinarySearch() bool {
    size := 32768

    arr := make([]int, size)
    SequenceBuilder.PackIncreasing(arr)

    if BinarySearch.Find(arr, -1) != size {
        return false
    }

    if BinarySearch.Find(arr, 2147483647) != size {
        return false
    }

    for i, v := range arr {
        if BinarySearch.Find(arr, v) != i {
            return false
        }
    }

    return true
}

func main() {
    TestRunner.ParseTest(testBinarySearch())
}

File: python/muse/algorithms/search/BinarySearchTest.py
from muse.algorithms.search import BinarySearch
from muse.util import SequenceBuilder
from muse.util import TestRunner

def testBinarySearch() -> bool:
    size = 32768

    arr = [0] * size
    SequenceBuilder.packIncreasing(arr)

    if BinarySearch.find(arr, -1) != size:
        return False

    if BinarySearch.find(arr, 2_147_483_647) != size:
        return False

    for i, v in enumerate(arr):
        if BinarySearch.find(arr, v) != i:
            return False

    return True

def main() -> None:
    TestRunner.parseTest(testBinarySearch())

if __name__ == "__main__":
    main()
